javaparser-core/src/main/java/com/github/javaparser/ast/expr/InstanceOfExpr.java

```diff
@@ -50,15 +53,22 @@
 
     private ReferenceType type;
 
+    @OptionalProperty
+    private SimpleName name;
```

Comment:
That seems like a simple and cheap solution. Nice!

Reply:
Cheers! 

A "proper" fix likely needs to make use of the `Pattern` (defined in the JEP, but not the JLS), but just to get something functional I was chuffed to get this far :)

Reply:
Once you start digging into this, you'll find that many decisions about how the AST is formed are arbitrary, so don't feel like there is a right way :-) This way it is definitely easy to use for the end user!